olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,199 @@
+            ),
+            "bits": PassConfigParam(
+                type_=int,
+                default_value=4,
+                description="quantization bits. Default value is 4",
+            ),
+            "dataset": PassConfigParam(
+                type_=str,
+                default_value="wikitext2",
+                description="Calibration dataset. Default value is wikitext2",
+            ),
+            "block_name_to_quantize": PassConfigParam(
+                type_=str,
+                default_value="model.layers",
+                description="Block name to quantize. Default value is model.layers.",
+            ),
+            "group_size": PassConfigParam(
+                type_=int,
+                default_value=128,
+                description="Block size for quantization. Default value is 128.",
+            ),
+            "batch_size": PassConfigParam(
+                type_=int,
+                default_value=1,
+                description="Batch size for quantization. Default value is 1.",
+            ),
+            "seed": PassConfigParam(
+                type_=int,
+                default_value=0,
+                description="Random seed for sampling calibration dataset. Default value is 0.",
+            ),
+            "damp_percent": PassConfigParam(
+                type_=float,
+                default_value=0.01,
+                description="Damping factor for quantization. Default value is 0.01.",
+            ),
+            "static_groups": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use static groups for quantization. Default value is False.",
+            ),
+            "true_sequential": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use true sequential for quantization. Default value is False.",
+            ),
+            "desc_act": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Use descriptive activation for quantization. Default value is False.",
+            ),
+            "sym": PassConfigParam(
+                type_=bool,
+                default_value=False,
+                description="Symmetric quantization. Default value is False.",
+            ),
+            "export_optimization": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Export optimization level. Default value is None.",
+            ),
+            "data_config": PassConfigParam(
+                type_=Union[DataConfig, Dict],
+                default_value=None,
+                description="""
+                    Data config for quantization. Default value is None.
+                """,
+            ),
+            "dataloader_func": PassConfigParam(
```

Comment:
>dataloader_func [](http://example.com/codeflow?start=13&length=15)

the dataloader_func seems duplicate with dataset param?

Reply:
I'm not sure about this. In my usage, dataset, data_config and dataloader_func may serve different usage.
dataset is usefull when user only want try a few already tested open source datasets, they only just pass a string,.
data_config can use huggingface dataset with different predifined strategy and process to use.
dataloader_func can be passed by user's predefined dataset function in their training code to easily use this olive pass